### PR TITLE
fix: ignore non matching logs earlier

### PR DESCRIPTION
**Pull request by jonaro00**

### CODE REVIEW

File: Cargo.lock
Comment (orhun) [inline Cargo.lock:1]: Are these `Cargo.lock` changes related?

Comment (jonaro00) [reply Cargo.lock:1]: Probably leftovers from a hasty merge of another PR

File: logger/src/lib.rs [hunk 1]
```diff
@@ -124,10 +124,14 @@ where
...
-                            debug!("stream receiver queue size {}", logs_rx.len())
+                        if !logs
+                            .first()
+                            .is_some_and(|l| l.deployment_id == deployment_id)
+                        {
+                            continue;
                         }
-
+                        // only if logs was for this recipient, to reduce spam.
+                        debug!("stream receiver queue size {}", logs_rx.len());
```
Comment (oddgrd) [inline logger/src/lib.rs:134]: We should still check if the rx queue is empty, I think.

Comment (jonaro00) [reply logger/src/lib.rs:134]: Wait.. haha the merge goofed everything

**Comment by jonaro00** [comment]
Can't prove that this will work, since deployer batchers could send mixed ids at intersection of deployments.

**Pull request closed without merging**